### PR TITLE
[#29] pattern extractor

### PR DESCRIPTION
**Pull request by pbroman**

### CODE REVIEW

File: kitodo-mediaserver-core/src/main/java/org/kitodo/mediaserver/core/processors/PatternExtractor.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    @Override
+    public String extract(String input) throws ConfigurationException {
+
+        if (CollectionUtils.isEmpty(regexList)) {
+            throw new ConfigurationException("There is no list of regular expressions for the pattern extractors, "
+                    + "please check your spring configuration.");
+        }
+
+        Optional<String> matchingRegex = regexList.stream()
+                .filter(regex -> input.matches(regex))
```
Comment (ponchofiesta) [inline kitodo-mediaserver-core/src/main/java/org/kitodo/mediaserver/core/processors/PatternExtractor.java]: IntelliJ says: `.filter(input::matches)` :)

**Comment by ponchofiesta** [comment]
Wouldn't it be more reliable to always parse the whole URL or path? For example numbers could be identically. At the moment the "0" is always 0. But what if it changes and there are two "/600/" or something similar? I thought of using one regex string and returning a List<String> with the results for the capture groups.

File: kitodo-mediaserver-core/src/main/java/org/kitodo/mediaserver/core/processors/PatternExtractor.java (hunk 1 above)
Comment (pbroman) [reply kitodo-mediaserver-core/src/main/java/org/kitodo/mediaserver/core/processors/PatternExtractor.java]: sounds good

**Comment by pbroman** [comment]
Of course, you can configure that as you wish. The present list of regex:s is just an example. In a production system I would indeed make the regex:s safer.

**Comment by ponchofiesta** [comment]
But IExtractor returns a String with the first capture group only. Or do you want to extend it later?

**Pull request merged**